catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/Poller.java

```diff
@@ -173,6 +178,30 @@ protected synchronized void pollCurrent(
       throw new IllegalStateException(message);
     }
 
+    if (!pollCurrentLock.tryLock()) {
+      final String message = "Multiple threads may not pollCurrent at the same time";
+      LOGGER.debug(message);
+      throw new UnsupportedOperationException(message);
```

Comment:
❓Could we use `IllegalStateException` instead, or would that cause problems?

Reply:
That's fine with me. I'll update it. Thanks for the suggestion.